### DISCUSSION TITLE
seaport js swap problem

### DISCUSSION THREAD

**Discussion by felixdev0429** [ProjectOpenSea/seaport-js#79] [Q&A] [answered]
https://github.com/ProjectOpenSea/seaport-js/tree/main/src/__tests__

I did tested this swap feature. but it is not working.
nft+erc20<=>nft+erc20 is working well. But nft+eth<=>nft+eth is not working.
Only nft swap with erc20 is working.
So I build new swap page simply using seaport js npm package and test it.
But same problem. in other words, both frontend and hardhat test(seaport js npm package) is same problem.
nft swap feature with erc20(nft + erc20 <=> nft + erc20) is working well. but with eth is not working.
If you give me this solution. it would be appreciated.
From opensea [partner.]
![Screenshot_1](https://user-images.githubusercontent.com/79739656/178448158-4b2ebd01-7af9-4097-9916-ba4778c4a477.png)
![Screenshot_2](https://user-images.githubusercontent.com/79739656/178448174-3417c5af-20e5-4292-8be4-470be5becf5b.png)

**Comment by sunwrobert** [comment, marked as answer]
You can't swap ETH on both sides. You can only swap ETH if you're the one fulfilling it. The other party can't offer ETH and you fulfill it

**Reply by felixdev0429** [reply]
Thanks for your reply.
So this feature I want is impossible?
Why can't offerer offer ETH when nft swap?
I confirmed it was working well in WETH(erc20) case.
Is it impossible in ETH?
Could you explain about that?
If you response about that, it would be appreciated.
Thanks again sir.

**Reply by sunwrobert** [reply]
Only one side can offer ETH and the one offering ETH has to be the one fulfilling the order. It's because ETH isn't an ERC-20 and you can't take ETH out of another person's wallet. It's the reason why people can only do regular offers on OpenSea in WETH/ERC-20, not ETH.

**Reply by felixdev0429** [reply]
Understood.
Thank you very much.
